server/src/MessageHandler.ts

```diff
@@ -5,77 +5,190 @@ import {
   Range,
   Location,
   DeclarationParams,
+  CompletionParams,
+  CompletionList,
+  CompletionItem,
+  Position,
 } from 'vscode-languageserver'
 import * as util from './util'
 import { fullDocumentRange } from './provider'
-import { getDMMF } from '@prisma/sdk'
-import { TextDocument, Position } from 'vscode-languageserver-textdocument'
+import { TextDocument } from 'vscode-languageserver-textdocument'
 import format from './format'
+import {
+  getSuggestionsForAttributes,
+  getSuggestionsForTypes,
+  getSuggestionForBlockTypes,
+  getSuggestionForFirstInsideBlock,
+  getSuggestionForSupportedFields,
+  getSuggestionsForInsideAttributes,
+} from './completions'
 
-function getWordAtPosition(document: TextDocument, position: Position): string {
-  const currentLine = document.getText({
-    start: { line: position.line, character: 0 },
-    end: { line: position.line, character: 9999 },
+function getCurrentLine(document: TextDocument, line: number) {
+  return document.getText({
+    start: { line: line, character: 0 },
+    end: { line: line, character: 9999 },
   })
+}
+
+function convertDocumentTextToTrimmedLineArray(
+  document: TextDocument,
+): Array<string> {
+  return [...Array(document.lineCount)].map((_, i) =>
+    getCurrentLine(document, i).trim(),
+  )
+}
+
+function isFirstInsideBlock(position: Position, currentLine: string): boolean {
+  if (currentLine.trim().length === 0) {
+    return true
+  }
+
+  const stringTillPosition = currentLine.slice(0, position.character)
+  const matchArray = new RegExp(/\w+/).exec(stringTillPosition)
```

Comment:
No need to call `new RegExp`.
https://jsbench.me/whkacl1gbx/1
```suggestion
  const matchArray = /\w+/.exec(stringTillPosition)
```